### DISCUSSION TITLE
Hello World on cheap 4 inch display

### DISCUSSION THREAD

**Discussion by jkaidor** [moononournation/Arduino_GFX#97] [Q&A]
Hello!
   I am trying to port a project from a 2.8 inch display to a 4-incher.   The processor card is a Teensy 4.1.  My interface card connects
SDO - aka "MISO" to the pin named D12 on the Teensy.  "MOSI" is connected to D11.   SCK is connected to pin D13.

Here's the Amazon link to the display:  https://www.amazon.com/gp/product/B09BZZXYJD

  As a first step, I am trying the "hello world" example program.  It DOES work on the 2.8 inch display with the following declarations:

Arduino_DataBus *bus = new Arduino_HWSPI(9 /* DC */, 10 /* CS */);
Arduino_GFX *gfx = new Arduino_ILI9341(bus, TFT_RST, 0 /* rotation */, false /* IPS */);

  However, it does NOT work on the four incher:

Arduino_DataBus *bus = new Arduino_HWSPI(9 /* DC */, 10 /* CS */);
Arduino_GFX *gfx = new Arduino_ST7796(bus, -1 /* RST */, 0 /* rotation */);

The display does light up, and there are random shimmering lines.  Blocks of lines are coming and going more or less when the hello world is trying to display its blocks of text.

The display RST is not controlled by the CPU - there are a couple of caps to give it a "reset pulse" on power up.

Can anybody supply a clue?

**Comment by jkaidor** [comment]
Simplifying it even further - I commented out all the graphics commands except for two lines:

    gfx->begin();
    gfx->fillScreen(BLACK);

...And I got a mostly black screen with shimmering white vertical lines.
Changing BLACK to GREEN, I got a mostly green screen with shimmering white vertical lines.
...This is starting to look like a bad display module. :(.

**Reply by moononournation** [reply]
I think it may not ST7796, same form factor red board can be ILI9488. Try using ILI9488_18bit to check is it work.

**Comment by jkaidor** [comment]
Thanks - I tried that, no joy.
Interestingly, I got a second board from the same Amazon vendor - that 
one works for video but the touch screen constantly reports being 
touched - even though I am not touching it.

I suspect that this particular vendor just has a pile of QC rejects.  A 
third board should
be coming in today.  Bet it also has some problem.

Will have to take advantage of Amazon's return policy.

                 - ***@***.***
